Saucier720-app/src/app/list/list.component.ts

```diff
@@ -1,123 +1,143 @@
 import { Component, OnInit } from '@angular/core';
-import { Ingredient } from '../core/interfaces/ingredient';
+import { Ingredient, List } from '../core/interfaces/ingredient';
 import { ListService } from '../core/services/list/list.service';
 import { HttpClient } from '@angular/common/http';
 import { HttpEvent, HttpEventType } from "@angular/common/http"
 import { lastValueFrom } from 'rxjs';
+import { AuthService } from '../core/services/Auth/auth.service';
+import { GetListRequest, PostListRequest, UpdateListRequest } from '../core/interfaces/types';
 
 @Component({
   selector: 'app-list',
   templateUrl: './list.component.html',
   styleUrls: ['./list.component.scss']
 })
 export class ListComponent implements OnInit {
-  ingredients: Array<Ingredient> = [];
-  newIngredientName: string = '';
-  newIngredientQuantity: number = 0;
+  name: string = '';
+  quantity: number = 0;
 
-  constructor(private listService: ListService, private http: HttpClient) { }
+  list: List = {
+    Ingredients: []
+  }
+
+  constructor(private listService: ListService, private http: HttpClient, private authService: AuthService) { }
 
   async ngOnInit() {
     await this.populateList();
   }
 
   public async populateList(): Promise<void> {
-    try {
-      const event: HttpEvent<any> = await lastValueFrom(this.listService.getList());
-      switch(event.type) {
-        case HttpEventType.Sent:
-          console.log('Request sent!');
-          break;
-        case HttpEventType.ResponseHeader:
-          console.log('Response header received!');
-          break;
-        case HttpEventType.DownloadProgress:
-          const kbLoaded = Math.round(event.loaded / 1024);
-          console.log(`Download in progress! ${kbLoaded}Kb loaded`);
-          break;
-        case HttpEventType.Response:
-          console.log('Done!', event.body);
-          this.ingredients = event.body;
-          break;
+    const request: GetListRequest = {
+      UserName: this.authService.getUsername(),
+    };
+    console.log('request: ', request);
+    this.listService.getList(request.UserName).subscribe({
+      next: (response: any) => {
```

Comment:
So clean

Reply:
I think we can even clamp this response type as well but I'm scared to try that...